Added Infinite Scroll

ege-del opened the pull request:
Added;
+ infinite scroll
+ card link fix
+ dna animation when page is loading
+ search bar shadow change
+ added axios library for XMLHttpRequests
+ simplified routes/index.js

ege-del committed:
infinite scroll

ege-del committed:
infinite scroll and fixes

ege-del commented:
to test page loading; 
quick hack to **double** items in collection
db.images.find({}).forEach(function(doc){var newDoc = doc; delete newDoc._id; db.images.insert(newDoc); })

adriantintpilver commented:
Hi @ege-del, thanks for your cooperation. 
I tested your version, the scroll loader works fine. Excelente work!!
The search works but does not maintain the value you search, in the last version the find characters remain when you arrive in the search page, and if you go to the profile of some image, and you go back with the back button, you return to the search page with the last search.
This is important, please, can you fix this version with the features we have in the previous version?

Many thanks!!
Regards!

ege-del committed:
checkpoint

ege-del committed:
Merge branch 'adriantintpilver:main' into main

ege-del commented:
Added pop-up window to browse without losing page info and kept single image page to make it easier to share.(similar implementation to Reddit)
Using url for page info and having back button causes other problems I think is the best of both worlds.

two actions to test;
click on image -> refresh page -> opens single image page
click on image -> click on pop-up title -> opens single image page

Also updated upload page design because bootstrap 5 broke it (needed bs5 for modals)
added commit for feedback, still needs some clean-up

adriantintpilver commented:
Hi Ege-del

I testing your version, and find some issues:

1) The file form is in Spanish now for the upload and for the edit.
2) In the image profile, don't show the video, the video frame doesn't appear.
3) when you search something, and you go to the result page, you lose the characters you use for the search, and in this page you don't know what the search condition is, that can be confusing. If you look at the previous version you see the "find" is passed around all the sites to keep the find characters you use in the last search.

On the other hand, I am very happy with this placeholder version keeping the title in a small version all the time, i think it is better.. You sure I am keeping this.

We are going in the right way, this is not an easy issue and you are doing a great job.

Many thanks for all the effort you put in this!!

ege-del commented:
thanks for the feedback !

for 1. I think this is a problem with bootstrap 5 and OS language, I haven't found anything about it online and my OS language is in English so I need your help on fixing that.
I'll commit the fixes for 2. and 3. tomorrow.

ege-del committed:
fix

adriantintpilver commented:
Hello @ege-del 

I was testing this version, and I saw her pretty well. I have only one thing to fix.
You show only the video if there is any video, and not show the image,  but I liked showing the image and the video.
I think not every time the image shows a piece of the video and we can allowed to show the image and the video.
Do you think it is possible to do this?

Tell me if you can or not and if not I'll take the post without problems. I will continue your job at this point, you've done a lot already.

Thanks a lot for your work in this.
Regards.

ege-del commented:
It's easy to add this feature but, I prefer keeping the design simple.
Maybe we can ask breck7 to decide ?

adriantintpilver commented:
I think it's fine yes, I tag it so you can see it. @Breck7

adriantintpilver commented:
Hello @Breck7, please give us your opinion on this point when you can, if you prefer that you can see image and video in detail or only video if there is and the image only if there is no video available. 
Thanks!

ege-del commented:
> Hello @breck7, please give us your opinion on this point when you can, if you prefer that you can see image and video in detail or only video if there is and the image only if there is no video available. Thanks!

I think this would be a good improvement. Dynamically showing image would keep the simplicity and the detailed image view,
for example adding a button to toggle the image.
Also, any updates on the spanish language problem ?

breck7 commented:
Simpler is better!
